### PR TITLE
146 hook auth logic

### LINKED ISSUE #146: Refactor: Hook auth logic into new client
# Tasks
- [ ] Login 
- [ ] Signup
- [ ] Error handling
- [ ] Verification code
- [ ] Forget password
- [ ] Signout logic
- [ ] Nav bar user logged in / not logged in

### PR DESCRIPTION
**Pull request by Thompson632**
Closes #146 

**Note: This logic does not affect existing functionality but has been tested with the existing server. You can change how the content is loaded by updating the [app.js]()**

## Changes
- Full authentication flow for:
  - Sign up
  - Sign in 
  - Verification code to verify the account
  - Resend verification code
  - Forgot username/password -> reset flow
 - Client / Server
   - Renamed the `userName` variable to be `username`
   - Formatting from vscode formatter
   - Fixed bug in server mongo config with URI
   - Added dev scripts for running outside of docker

### CODE REVIEW

File: client/app.js [hunk 1]
```diff
@@ -17,21 +19,12 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
 app.use(cookieParser());
 // TODO: Uncomment 'public/new' for new client
-app.use(express.static(path.join(__dirname, "public")));
-// app.use(express.static(path.join(__dirname, "public/new")));
+// app.use(express.static(path.join(__dirname, "public")));
```
Comment (Thompson632) [inline client/app.js:22]: Revert before merge

File: server/server.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
   useNewUrlParser: true,
-  useUnifiedTopology: true,
-}); //mongoose.connect(DB_URI); is what it used to be. this did not work
+  useUnifiedTopology: true
+});
 
-app.listen(3000, '0.0.0.0', () => {
-    console.log("Server running on port 3000");
-    console.log("_____________________________");
+app.listen(3001, '0.0.0.0', () => {
```
Comment (Thompson632) [inline server/server.js:12]: Revert before merge

File: client/public/new/static/js/login.js
Comment (nicksonsini97) [inline client/public/new/static/js/login.js:1]: I think line 72 should be a ":" instead of a ","

File: client/public/new/static/js/login.js [hunk 3]
```diff
@@ -1,60 +1,205 @@
...
+
+    const verificationCode = await validationHandler.getVerificationCode();
+    emailWrapper.sendEmail(fullName, email, verificationCode, emailjs, "newUser", username);
+
+    const user = {
+      fullName: fullName,
+      username: username,
+      password: password,
+      email: email,
+      verificationCode, verificationCode,
```
Comment (nicksonsini97) [inline client/public/new/static/js/login.js:72]: this line

**Review by nicksonsini97 — APPROVED** [review]
This all looks good to me. I saw one small typo in the login.js file but overall I'm good with everything!

File: server/server.js (hunk 2 above)
Comment (Thompson632) [reply server/server.js]: c

File: client/public/new/static/js/login.js (hunk 3 above)
Comment (Thompson632) [reply client/public/new/static/js/login.js]: just fixed

File: client/public/new/static/js/login.js
Comment (Thompson632) [reply client/public/new/static/js/login.js:1]: fixed

**Pull request merged**